pynamodb/models.py

```diff
@@ -169,40 +171,104 @@ class MetaModel(AttributeContainerMeta):
     """
     def __init__(cls, name, bases, attrs):
         super(MetaModel, cls).__init__(name, bases, attrs)
+
+        # We don't' allow inheritance from multiple direct parents
+        if (len(bases) > 1):
+            raise InheritanceError("Model can't inherit from multiple direct parents")
+
+        parent = None if len(bases) == 0 else bases[0]
+        # TODO: special parents as constants
+        parent_is_attribute_container = parent.__name__ == 'AttributeContainer'
+        parent_is_model = parent.__name__ == 'Model'
+        parent_meta_class = None
+        is_parent_abstract = None
+        if parent  and not parent_is_attribute_container:
+            parent_meta_class = getattr(parent, META_CLASS_NAME, None)
+        if parent_meta_class and hasattr(parent_meta_class, ABSTRACT):
+            is_parent_abstract = getattr(parent_meta_class, ABSTRACT)
+
+        # Temporary check until implementing other inheritance types
+        if parent and not parent_is_attribute_container and not parent_is_model and not is_parent_abstract:
+            raise NotImplementedError("Inheritance from non abstract models is not supported now")
+
         if isinstance(attrs, dict):
+            meta_class = attrs.get(META_CLASS_NAME, None)
+
+            # Check special parents to avoid copying DefaultMeta
+            if parent and not parent_is_attribute_container and not parent_is_model:
+                if not meta_class:
+                    # Prepare an empty meta class
+                    meta_class = type(META_CLASS_NAME, (), {})
+                    setattr(cls, META_CLASS_NAME, meta_class)
+
+                if parent_meta_class:
+                        # Copying missing meta attributes from parent(s)
```

Comment:
If we are going to go this route, it seems like the easier thing would just be to rely on python's inheritance rules and access the meta class through the passed-in attributes.

Reply:
Hi @jpinner-lyft,
If you choose this approach please don't forget that model inheriting abstract model will also be abstract.
I believe you should handle it somehow.